examples/inverse_kinematics/README.md

```diff
@@ -0,0 +1,202 @@
+# IK solutions service
+
+## Introduction
+
+In this example, we will explore the utilization of ROS2 for deriving Inverse Kinematics (IK) solutions for specified tool poses. The process involves the following steps:
+
+**Solution Identification:** When the system successfully computes an IK solution, a Transform (TF) is broadcasted to RViz. This event is clearly indicated by tagging the TF with a "YES" marker.
+
+**No Solution Handling:** In scenarios where the IK solution is not found, the TF is still published to RViz but is distinctly marked with a "NO".
+
+**Robot Movement Execution:** Upon finding a viable IK solution, the robot is commanded to relocate to the newly determined position.
+
+**Alternative Strategy for Unsolvable Positions:** If an IK solution is unachievable, the system adopts a contingency strategy. It instructs the robot to approximate the target position as closely as feasible. This is done by allowing modifications in the robot's yaw (rotational movement) and height, thus optimizing the robot's position relative to the desired pose.
+
+This approach ensures a robust and adaptive response to varying tool pose requests, enhancing the flexibility and efficiency of the robotic system in dynamic environments.
+
+The given Python script will:
+- undock Spot;
+- move it forward approximatively 1m;
+- rotate it 90 degrees counter-clockwise;
+- search and test IK solutions for random tool poses;
+- dock Spot back to the given docking station.
+
+## Running the Example
+
+1. Make sure the robot is docked, and there is approximately 2 meters free in front of the docking station.
+
+2. Make sure you've built and sourced your workspace:
+   ```bash
+   cd <ros2 workspace>
+   colcon build --symlink-install
+   source /opt/ros/humble/setup.bash
+   source ./install/local_setup.bash
+   ```
+3.  Start the driver:
+    ```bash
+    bdai spot driver spot_name --launch-rviz
```

Comment:
Users of this repo won't necessarily have access to the `bdai` command line tool

Reply:
Updated. I added instructions that do not require bdai repo or bdai docker container.